dso-l2/src/main/java/com/tc/objectserver/persistence/FlatFileStorageServiceProvider.java

```diff
@@ -72,7 +77,9 @@ public boolean initialize(ServiceProviderConfiguration configuration) {
     if (!this.shouldPersistAcrossRestarts) {
       file.delete();
     }
-    return configuration.getServiceType().cast(new FlatFilePersistentStorage(file));
+    FlatFilePersistentStorage storage = new FlatFilePersistentStorage(file);
+    storages.add(storage);
```

Comment:
Will this be sufficient?  If we try to clear() before all the entities have started (which is always the case), their files will still be on disk.

Or do we ensure that these are deleted when the entity is first created (or otherwise first tries to open the file)?